File: frontend/src/types/forecast.ts
export interface Forecast {
  id: string;
  user_id: string;
  market_id: string;
  outcome_id: string;
  points: number;
  reward_amount?: number | null; // Actual reward amount if forecast wins
  status: 'pending' | 'won' | 'lost';
  is_flagged: boolean;
  created_at: string;
  updated_at: string;
  outcome_name?: string;
  market_title?: string;
  market_status?: string;
}

export interface ForecastCreate {
  outcome_id: string;
  points: number;
}

export interface ForecastUpdate {
  outcome_id?: string;
  points?: number;
}

export interface ForecastResponse {
  success: boolean;
  data: {
    forecast: Forecast;
    new_balance: number;
    updated_outcome?: {
      id: string;
      name: string;
      total_points: number;
    };
  };
  message: string;
}

export interface ForecastListResponse {
  success: boolean;
  data: {
    forecasts: Forecast[];
    user_forecast?: Forecast | null;
    pagination: {
      page: number;
      limit: number;
      total: number;
      pages: number;
    };
  };
}

File: backend/app/schemas/forecast.py
"""
Forecast schemas
"""
from pydantic import BaseModel, Field, validator
from typing import Optional
from datetime import datetime


class ForecastBase(BaseModel):
    """Base forecast schema"""
    outcome_id: str = Field(..., description="ID of the outcome to forecast")
    points: int = Field(..., ge=20, description="Number of chips to allocate (minimum 20)")


class ForecastCreate(ForecastBase):
    """Forecast creation schema"""
    pass


class ForecastUpdate(BaseModel):
    """Forecast update schema"""
    outcome_id: Optional[str] = None
    points: Optional[int] = Field(None, ge=20)


class ForecastResponse(BaseModel):
    """Forecast response schema"""
    id: str
    user_id: str
    market_id: str
    outcome_id: str
    points: int
    reward_amount: Optional[int] = None  # Actual reward amount when forecast wins (null if pending or lost)
    status: str  # pending, won, lost
    is_flagged: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class ForecastDetailResponse(ForecastResponse):
    """Forecast detail response with related data"""
    outcome_name: Optional[str] = None
    market_title: Optional[str] = None


class ForecastListResponse(BaseModel):
    """Forecast list response schema"""
    forecasts: list[ForecastDetailResponse]
    total_count: int
    page: int
    limit: int
    total_pages: int

